controllers/gce/controller/controller.go

```diff
@@ -53,17 +54,45 @@ var (
 	storeSyncPollPeriod = 5 * time.Second
 )
 
+// ControllerContext holds
+type ControllerContext struct {
+	IngressInformer cache.SharedIndexInformer
+	ServiceInformer cache.SharedIndexInformer
+	PodInformer     cache.SharedIndexInformer
+	NodeInformer    cache.SharedIndexInformer
+	// Stop is the stop channel shared among controllers
+	StopCh chan struct{}
+}
+
+func NewControllerContext(kubeClient kubernetes.Interface, namespace string, resyncPeriod time.Duration) *ControllerContext {
+	return &ControllerContext{
+		IngressInformer: informerv1beta1.NewIngressInformer(kubeClient, namespace, resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}),
+		ServiceInformer: informerv1.NewServiceInformer(kubeClient, namespace, resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}),
+		PodInformer:     informerv1.NewPodInformer(kubeClient, namespace, resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}),
+		NodeInformer:    informerv1.NewNodeInformer(kubeClient, resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}),
```

Comment:
Do you need to resync all four resources?

Reply:
Also why don't you use the informerFactory?

Reply:
Because informerFactory does not support specifying namespace.

@nicksardo correct me if I am wrong, ingress controller relies on the informer to trigger resync.

Reply:
Yes, the controller relies on the the resync periods to verify things are correct.